src/index.ts

```diff
@@ -8,32 +8,100 @@ import {
-            { Name: "Environment", Value: options.environment },
-            { Name: "Project", Value: project },
-          ],
-          Unit: "Count",
-          Value: 1,
-        },
-      ],
-      Namespace: "cuckoo",
-    };
-    await client.send(new PutMetricDataCommand(input));
-    console.log("Metric published to CloudWatch");
+    try {
+      const deployDuration = Number.parseInt(options.deployDuration);
+      if (Number.isNaN(deployDuration)) {
+        throw new TypeError("Deploy duration must be a number");
+      }
+
+      if (!VALID_STATUSES.has(options.status)) {
+        throw new TypeError("Invalid status");
+      }
+
+      const input: PutMetricDataCommandInput = {
+        MetricData: [
+          {
+            MetricName: `deployment.${options.status}`,
+            Dimensions: [
+              { Name: "Environment", Value: options.environment },
+              { Name: "Project", Value: project },
+              { Name: "Status", Value: options.status },
+            ],
+            Unit: "Count",
+            Value: 1,
+          },
+          {
+            MetricName: `deployment.${options.status}`,
+            Dimensions: [{ Name: "Environment", Value: options.environment }],
+            Unit: "Count",
+            Value: 1,
+          },
+          {
+            MetricName: `deployment.${options.status}`,
+            Dimensions: [{ Name: "Project", Value: project }],
+            Unit: "Count",
+            Value: 1,
+          },
+          {
+            MetricName: `deployment.${options.status}`,
+            Dimensions: [{ Name: "Status", Value: options.status }],
+            Unit: "Count",
+            Value: 1,
+          },
+          {
+            MetricName: `deployment.${options.status}`,
+            Unit: "Count",
+            Value: 1,
+          },
+          {
+            MetricName: "deployment.duration",
+            Dimensions: [
+              { Name: "Environment", Value: options.environment },
+              { Name: "Project", Value: project },
+            ],
+            Value: deployDuration,
+            Unit: "Seconds",
+          },
+          {
+            MetricName: "deployment.duration",
+            Dimensions: [{ Name: "Project", Value: project }],
+            Value: deployDuration,
+            Unit: "Seconds",
+          },
+          {
+            MetricName: "deployment.duration",
+            Dimensions: [{ Name: "Environment", Value: options.environment }],
+            Value: deployDuration,
+            Unit: "Seconds",
+          },
+          {
+            MetricName: "deployment.duration",
+            Value: deployDuration,
+            Unit: "Seconds",
+          },
+        ],
+
+        Namespace: "cuckoo",
+      };
+
+      await client.send(new PutMetricDataCommand(input));
+      console.log("Deployment metrics published to CloudWatch");
+    } catch (error) {
+      console.error("Error sending metrics to CloudWatch:", error);
```

Comment:
Perhaps make the script exit with the correct exit codes in these scenarios (process.exit)?